packages/manager/modules/vrack/src/dashboard/vrack.controller.js

```diff
@@ -264,82 +268,366 @@ export default class VrackMoveDialogCtrl {
     );
   }
 
-  getAllowedServices() {
-    return this.OvhApiVrack.Aapi()
-      .allowedServices({ serviceName: this.serviceName })
-      .$promise.then((allServicesParam) => {
-        let allServices = {
-          ...allServicesParam,
-          dedicatedCloud: allServicesParam.dedicatedCloud.filter((service) => {
-            return (
-              VrackMoveDialogCtrl.isServiceAllowed(service) &&
-              service.productReference === 'EPCC'
-            );
-          }),
-          managedBareMetal: allServicesParam.dedicatedCloud.filter(
-            (service) => {
-              return (
-                VrackMoveDialogCtrl.isServiceAllowed(service) &&
-                service.productReference === 'MBM'
-              );
-            },
-          ),
-          dedicatedServer: allServicesParam.dedicatedServer.filter(
-            VrackMoveDialogCtrl.isServiceAllowed,
-          ),
-          dedicatedServerInterface: allServicesParam.dedicatedServerInterface.filter(
-            ({ dedicatedServer }) =>
-              VrackMoveDialogCtrl.isServiceAllowed(dedicatedServer),
-          ),
-        };
-        allServices = mapValues(allServices, (services, serviceType) => {
-          if (isArray(services)) {
-            return map(services, (service) =>
-              this.constructor.fillServiceData(serviceType, service),
-            );
-          }
-          return services;
-        });
+  retrieveEligibleServices() {
+    this.data.eligibleServices = null;
+    this.loaders.eligibleServices = true;
+    this.getEligibleServices();
+  }
 
-        // We need to append dedicatedServerInterfaces list to dedicatedServers list.
-        if (
-          has(allServices, 'dedicatedServerInterface') &&
-          allServices.dedicatedServerInterface.length > 0
-        ) {
-          // If dedicatedServers list doesn't exist, we create it first.
-          if (!has(allServices, 'dedicatedServer')) {
-            allServices.dedicatedServer = [];
-          }
+  // Get service info for each cloud project retrieved from API
+  updateCloudProjectServiceInfo() {
```

Comment:
Most of your functions `update` could be moved to the service file to reduce the size of the controller and also to improve reusability

Reply:
We need to display partial result which could not be possible if I put it on service file.